### PR TITLE
Refactor/90

### PR DESCRIPTION
**Pull request by disog**
The help message option filters were moved to a property of the formatter configuration.

Various tests were added to cover edge cases.

### CODE REVIEW

File: packages/tsargp/test/parser/parser.completion.spec.ts [hunk 1]
```diff
@@ -403,39 +376,32 @@ describe('ArgumentParser', () => {
...
           type: 'strings',
           names: ['-ss'],
-          complete: vi.fn().mockImplementation(() => {
-            throw 'abc';
-          }),
+          complete: vi.fn().mockImplementation(() => []),
         },
       } as const satisfies Options;
       const parser = new ArgumentParser(options);
-      await expect(parser.parse('cmd -ss ', { compIndex: 8 })).rejects.toThrow(/^$/);
```
Comment (disog) [inline packages/tsargp/test/parser/parser.completion.spec.ts:437]: Preserve this expectation

**Review by disog — COMMENTED** [review]
LGTM
